### ISSUE TITLE
Using YOLOv5 on the exDark Dataset

### ISSUE THREAD

**Issue by TobiasSchotter** [ultralytics/yolov5#7983]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I am currently trying to use YOLOv5 on the exDark Dataset and I was wondering if I had to convert all the label files to the xywh-format? Same for the labels to indexes? Or is there a way to use that dataset out of the box? (https://github.com/cs-chan/Exclusively-Dark-Image-Dataset/tree/master/Groundtruth).

I structures the data in the form:

📦exDark
 ┣ 📂test
 ┣ 📂train
 ┣ 📂val
 ┃ ┗ 📂images
 ┃ ┗ 📂labels
      ┗ 📜2015_00001.txt
       ....

And I removed the first line in each .txt File:

Example: 
[1] `% bbGt version=3` 
[2-n] `Bicycle 204 28 271 193 0 0 0 0 0 0 0`

During training I get:

`dataset = LoadImagesAndLabels( File " {PATH} \YOLOv5-vs-RCNN\yolov5\utils\dataloaders.py", line 465, in __init__ labels, shapes, self.segments = zip(*cache.values()) ValueError: not enough values to unpack (expected 3, got 0)
`

**Comment by glenn-jocher** [comment]
@TobiasSchotter hi there,

You will indeed need to convert the labels to YOLOv5 compatible format (i.e., xywh-format) for the exDark Dataset. You can use our provided conversion script using the Ultralytics Docs at https://docs.ultralytics.com/yolov5/. This will help you convert the labels to indexes. Let us know if you need any further assistance!

Best regards

**Comment by BeanBagKing** [comment]
@glenn-jocher - I know this is a little old, but can you be more specific about the location of that conversion script? There's only one mention of "script" on that page, and that's for training. No mention of conversion, the exdark dataset, or bbGt/Bounding box (bb) annotations struct (which I only think is the name of that structure). Nor do I see any script on the yolov5 github (here) that looks like a conversion script, sorry if I missed it. I also tried using roboflow, but it doesn't seem to understand that annotation format (I both tried it, and checked in the listed formats and didn't see it, but I could be doing something wrong).

**Comment by glenn-jocher** [comment]
@BeanBagKing you can create a custom script to convert the exDark annotations to YOLO format. Here's a basic example:

```python
def convert_to_yolo_format(input_file, output_file, class_mapping):
    with open(input_file, 'r') as f:
        lines = f.readlines()[1:]  # Skip the first line
    with open(output_file, 'w') as f:
        for line in lines:
            parts = line.split()
            class_name = parts[0]
            bbox = list(map(float, parts[1:5]))
            class_id = class_mapping[class_name]
            # Convert bbox to YOLO format (normalized xywh)
            x_center = (bbox[0] + bbox[2]) / 2
            y_center = (bbox[1] + bbox[3]) / 2
            width = bbox[2] - bbox[0]
            height = bbox[3] - bbox[1]
            f.write(f"{class_id} {x_center} {y_center} {width} {height}\n")

# Example usage
class_mapping = {'Bicycle': 0}  # Add all class mappings
convert_to_yolo_format('input.txt', 'output.txt', class_mapping)
```

Adjust `class_mapping` with your dataset's classes. This should help you convert the annotations.

**Comment by BeanBagKing** [comment]
Thank you.

For anyone else coming across this via Google like myself, here's what I ended up using: https://github.com/BeanBagKing/YOLO_Training/blob/main/OtherScripts/exdark_converter.py

Doing a bunch of spot checking, it looks like it converted everything nearly correctly. When importing into Roboflow, I got a warning that several bounding boxes went outside the image area and Roboflow corrected those. I looked at a few of those images and nothing was wrong with them. There was at least one image that did not have anything annotated, and at least one image I came across that had the wrong annotation for an object. I think this is an issue with the original dataset though, not in the conversion.
